MonoGame.Framework.Content.Pipeline/Graphics/MeshHelper.cs

```diff
@@ -373,16 +374,75 @@ public static IList<BoneContent> FlattenSkeleton(BoneContent skeleton)
             return results;
         }
 
+        /// <summary>
+        /// Merge any positions in the <see cref="PositionCollection"/> of the
+        /// specified mesh that are at a distance less than the specified tolerance
+        /// from each other.
+        /// </summary>
+        /// <param name="mesh">Mesh to be processed.</param>
+        /// <param name="tolerance">Tolerance value that determines how close 
+        /// positions must be to each other to be merged.</param>
+        /// <remarks>
+        /// This method will also update the <see cref="VertexContent.PositionIndices"/>
+        /// in the <see cref="GeometryContent"/> of the specified mesh.
+        /// </remarks>
         public static void MergeDuplicatePositions(MeshContent mesh, float tolerance)
         {
-            throw new NotImplementedException();
+            for (var i = mesh.Positions.Count - 1; i >= 1; i--)
+            {
+                var pi = mesh.Positions[i];
+                for (var j = i - 1; j >= 0; j--)
+                {
+                    var pj = mesh.Positions[j];
+                    if (Vector3.Distance(pi, pj) <= tolerance)
+                    {
+                        UpdatePositionIndices(mesh, i, j);
+                        mesh.Positions.RemoveAt(i);
+                        break;
+                    }
+                }
+            }
         }
 
+        /// <summary>
+        /// Merge vertices with the same <see cref="VertexContent.PositionIndices"/> and
+        /// <see cref="VertexChannel"/> data within the specified
+        /// <see cref="GeometryContent"/>.
+        /// </summary>
+        /// <param name="geometry">Geometry to be processed.</param>
         public static void MergeDuplicateVertices(GeometryContent geometry)
         {
-            throw new NotImplementedException();
+            var verts = geometry.Vertices;
+            for (var i = verts.VertexCount - 1; i >= 1; i--)
+            {
+                for (var j = i - 1; j >= 0; j--)
+                {
+                    if (verts.PositionIndices[i] != verts.PositionIndices[j])
+                        continue;
+
+                    if (verts.Channels.Any(channel => channel[i] != channel[j]))
+                        continue;
+
+                    // vertices are equal, so let's merge them
+                    for (var index = 0; index < geometry.Indices.Count; index++)
+                    {
+                        if (geometry.Indices[index] == i)
+                            geometry.Indices[index] = j;
+                    }
+
+                    verts.RemoveAt(i);
+                    break;
+                }
+            }
```

Comment:
It's much easier to check and remove duplicate positions right in the CreatePosition(). Who knows, they have to implement it in MeshHelper anyways, so why having duplicate code.
As for merging vertices, does it discard repeated triangles or something? I really don't get it.

> That would be great!

I don't have anything that works with tolerance/distance though... give me some time.